lib/order.rb

```diff
@@ -0,0 +1,87 @@
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  @@tax_rate = 0.075
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    unless fulfillment_status == :pending ||
+           fulfillment_status == :paid ||
+           fulfillment_status == :processing ||
+           fulfillment_status == :shipped ||
+           fulfillment_status == :complete
+      raise ArgumentError, "invalid fulfillment status"
+    end
+  end
+
+  def self.products_hash(products_string)
+    items_with_prices = products_string.split(";")
+
+    products_hash = {}
+
+    items_with_prices.each do |item_with_price|
+      array = item_with_price.split(":")
+
+      item = array[0]
+      price = array[1].to_f
+
+      products_hash[item] = price
+    end
+
+    return products_hash
+  end
+
+  def self.all
+    orders_array = []
+
+    CSV.open("data/orders.csv", "r").each do |line|
+      id = line[0].to_i
+      products_string = line[1]
+      products = products_hash(products_string)
+      customer_id = line[2].to_i
+      customer = Customer.find(customer_id)
+      fulfillment_status = line[3].to_sym
+      orders_array << Order.new(id, products, customer, fulfillment_status)
+    end
+
+    return orders_array
+  end
+
+  def self.find(id)
+    Order.all.each { |order| return order if order.id == id }
+    return nil
+  end
+
+  def tax_rate
+    @@tax_rate
+  end
+
+  def costs
+    costs = products.values
+  end
+
+  def total
+    total = (costs.sum + costs.sum * tax_rate).round(2)
```

Comment:
When you run `rake`, Ruby/Minitest will give you a warning saying that the variable `total` here (as in `total = (costs.sum ... `) isn't being used, and can therefore probably deleted/re-written as such:

```ruby
def total
  (costs.sum + costs.sum * tax_rate).round(2)
end
```

if you wanted (or it can have the explicit `return`)